src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -163,6 +178,23 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
       return;
     }
 
+    String messageText = request.getParameter("message");
+    Pattern r = Pattern.compile("\\[(.+?)\\]");
+    Matcher m = r.matcher(messageText);
```

Comment:
Its a bad convention in Java to use one-lettered variable names. I know it can be cumbersome sometimes, but have names that will let the reader know what the object is anywhere else in the code. So for `m` it could be something like `messageTextMatcher`.

Same for other variables.